packages/mobile/src/components/fields/PickArtworkField.tsx

```diff
@@ -87,23 +94,35 @@ export const PickArtworkField = (props: PickArtworkFieldProps) => {
         },
         source: 'original'
       })
-      setIsLoading(true)
+      setIsAutogenerated(false)
+      setStatus({ imageLoading: true })
     }
     launchSelectImageActionSheet(handleImageSelected, secondary)
-  }, [secondary, setValue])
+  }, [secondary, setArtwork, setIsAutogenerated, setStatus])
 
   const source = useMemo(() => ({ uri: trackArtworkUrl }), [trackArtworkUrl])
 
+  const generatePlaylistArtwork = useGeneratePlaylistArtwork(collectionId)
+
+  const handleRemoveArtwork = useCallback(async () => {
+    setStatus({ imageLoading: true, imageGenerating: true })
+    const { file, url } = await generatePlaylistArtwork()
+    setArtwork({ url, file })
+    setIsAutogenerated(true)
+  }, [setStatus, generatePlaylistArtwork, setArtwork, setIsAutogenerated])
+
   return (
     <View style={styles.root}>
       <DynamicImage
         source={source}
-        onLoad={() => setIsLoading(false)}
+        onLoad={() =>
+          setStatus({ imageLoading: false, imageGenerating: false })
+        }
         style={styles.image}
         noSkeleton
       >
         <View style={styles.iconPicture}>
-          {isLoading ? (
+          {status.imageLoading ? (
```

Comment:
does this cover the case of status.imageGenerating as well?

Reply:
no great call